iotilecore/iotile/core/hw/reports/signed_list_format.py

```diff
@@ -23,6 +29,41 @@ class SignedListReport(IOTileReport):
     def __init__(self, rawreport, **kwargs):
         super(SignedListReport, self).__init__(rawreport, signed=True, encrypted=False, **kwargs)
 
```

Comment:
I would recommend actually just making these static members inside the class:

```python

class SignedListReport:
    SIGNED_WITH_HASH = 0
    SIGNED_WITH_USER_KEY = 1
    SIGNED_WITH_DEVICE_KEY = 2

    _KEY_TYPE_MAP = {
        AuthProvider.NoKey: ReportSignatureFlags.SIGNED_WITH_HASH,
        ...
    }

    @classmethod
    def KeyTypeToStreamType(cls, key_type):
        stream_type = cls._KEY_TYPE_MAP.get(key_type)
        if stream_type is None:
            raise ArgumentError("Unknown Key type")
       return stream_type

# Same with the reverse direction
```

Reply:
You then don't need to export another public enum and can access the values as needed on the class itself.

Reply:
The plan was to have a public enum, so it can be used outside of the class, for example in the tests, not sure how better to do. 
It looks like I should not have broken the compability between signature flags and key types but just document in explicitly